### PR TITLE
Update order email payload

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds more details to the order email payload.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/order/notifications.py [hunk 1]
```diff
@@ -16,16 +17,82 @@
...
+    if first_image:
+        first_image_payload = ({"original": get_image_payload(first_image)},)
+    images_payload = None
+    if images:
+        images_payload = [{"original": get_image_payload(image) for image in images}]
+    return {"first_image": first_image_payload, "images": images_payload}
+
+
+def get_product_attributes(product):
+    attributes = product.attributes.all()
```
Comment (IKarbowiak) [inline saleor/order/notifications.py:40]: Maybe some `prefetch_related`?

File: saleor/order/notifications.py [hunk 2]
```diff
@@ -16,16 +17,82 @@
...
+                    }
+                },
+                "values": [
+                    {
+                        "name": value.name,
+                        "value": value.value,
+                        "slug": value.slug,
+                        "file_url": value.file_url,
+                    }
+                    for value in attr.values.all()
```
Comment (IKarbowiak) [inline saleor/order/notifications.py:58]: Maybe we can use `values` here?
Like:
```suggestions
for value in attr.values.values("name", "value", "slug", "file_url")
```

**Review by fowczarek — APPROVED** [review]

File: saleor/plugins/email_common.py [hunk 3]
```diff
@@ -139,6 +142,32 @@ def format_address(this, address, include_phone=True, inline=False, latin=False)
...
+        date_format = "%d-%m-%Y"
+    return date.strftime(date_format)
+
+
+def get_product_image_thumbnail(this, size, image):
+    expected_size = get_thumbnail_size(size, "thumbnail", "products", on_demand=False)
+    return image["original"][expected_size]
+
+
+def compare(this, val1, compare_operator, val2):
```
Comment (maarcingebala) [inline saleor/plugins/email_common.py:157]: This and other template helpers could use docstrings.

File: saleor/order/notifications.py [hunk 4]
```diff
@@ -16,16 +17,82 @@
...
+
+def get_default_images_payload(images: List[ProductImage]):
+    first_image_payload = None
+    first_image = images[0] if images else None
+    if first_image:
+        first_image_payload = ({"original": get_image_payload(first_image)},)
+    images_payload = None
+    if images:
+        images_payload = [{"original": get_image_payload(image) for image in images}]
+    return {"first_image": first_image_payload, "images": images_payload}
```
Comment (maarcingebala) [inline saleor/order/notifications.py:36]: Do we need to return the first image separately? `images` contains it as well.

File: saleor/order/notifications.py (hunk 1 above)
Comment (korycins) [reply saleor/order/notifications.py:40]: It is already prefetched in `get_default_order_payload`

File: saleor/order/notifications.py (hunk 4 above)
Comment (korycins) [reply saleor/order/notifications.py:36]: We were thinking about this, but the first_image gives you an easier interface to build email content

Comment (maarcingebala) [reply saleor/order/notifications.py:36]: 👍

**Review by maarcingebala — APPROVED** [review]

**Review by d-wysocki — APPROVED** [review]

File: saleor/order/notifications.py (hunk 1 above)
Comment (patrys) [reply saleor/order/notifications.py:40]: Note: I want all prefetches to go away in favor of data loaders eventually. Please don't add more prefetches 😄

Comment (korycins) [reply saleor/order/notifications.py:40]: I will create the task for this as for now I want to finalize this feature first.

**Pull request merged**